metrics_api/lib/opentelemetry/metrics/meter.rb

```diff
@@ -56,11 +56,11 @@ def create_observable_up_down_counter(name, unit: nil, description: nil, callbac
       private
 
       def create_instrument(kind, name, unit, description, callback)
-        raise InstrumentNameError if name.nil?
-        raise InstrumentNameError if name.empty?
-        raise InstrumentNameError unless NAME_REGEX.match?(name)
-        raise InstrumentUnitError if unit && (!unit.ascii_only? || unit.size > 63)
-        raise InstrumentDescriptionError if description && (description.size > 1023 || !utf8mb3_encoding?(description.dup))
+        raise InstrumentNameError if invalid_name?(name)
+        raise InstrumentUnitError if invalid_unit?(unit)
+        raise InstrumentDescriptionError if invalid_description?(description)
+
+        name = normalize_name(name)
```

Comment:
This is new. From the Metrics' Meter API specification for the [Instrument Name Syntax](https://opentelemetry.io/docs/reference/specification/metrics/api/#instrument-name-syntax) it says:

> They are case-insensitive, ASCII strings.

So here I'm normalizing the name with `downcase` before using it in `@instrument_registry[name]`